### PR TITLE
feat(anvil): remove all txs from tx pool by sender origin

### PR DESCRIPTION
**Pull request by PanGan21**
## Motivation
Context: #7250 

## Solution

Adds `anvil_removePoolTransactions` method that removes pool transactions by sender origin.

### CODE REVIEW

File: crates/anvil/src/eth/pool/mod.rs [hunk 1]
```diff
@@ -342,6 +347,31 @@ impl PoolInner {
...
+        tx_hashes.extend(
+            self.ready_transactions
+                .transactions_by_sender(sender)
+                .map(|tx| tx.hash())
+                .collect::<Vec<TxHash>>(),
+        );
+
+        if tx_hashes.is_empty() {
+            return vec![]
+        }
```
Comment (mattsse) [inline crates/anvil/src/eth/pool/mod.rs:364]: can we extract this to a `transactions_by_sender` that merges those?

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, one suggestion

File: crates/anvil/src/eth/pool/mod.rs (hunk 1 above)
Comment (PanGan21) [reply crates/anvil/src/eth/pool/mod.rs]: Refactored! Let me know if this is what you meant!

**Review by mattsse — APPROVED** [review]
ty

**Pull request merged**